src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/writer/CatalogMultiResourceItemWriter.java

```diff
@@ -0,0 +1,225 @@
+import org.springframework.batch.core.StepExecution;
+import org.springframework.batch.item.ExecutionContext;
+import org.springframework.batch.item.ItemStreamException;
+import org.springframework.batch.item.file.MultiResourceItemWriter;
+import org.springframework.batch.item.file.ResourceAwareItemWriterItemStream;
+import org.springframework.batch.item.file.ResourceSuffixCreator;
+import org.springframework.batch.item.file.SimpleResourceSuffixCreator;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+import org.springframework.util.Assert;
+
+/**
+ * Wraps a {@link ResourceAwareItemWriterItemStream} and creates a new output resource when the count of items written in current resource exceeds
+ * {@link #setItemCountLimitPerResource(int)}. Suffix creation can be customized with {@link #setResourceSuffixCreator(ResourceSuffixCreator)}.
+ * 
+ * Note that new resources are created only at chunk boundaries i.e. the number of items written into one resource is between the limit set by
+ * {@link #setItemCountLimitPerResource(int)} and (limit + chunk size).
+ * 
+ * Code based on original {@link MultiResourceItemWriter} by Robert Kasanicky.
+ * 
+ * @param <T> item type
+ * 
+ * @author Robert Kasanicky
+ * @author Alessio Fabiani, GeoSolutions
+ */
+public class CatalogMultiResourceItemWriter<T> extends CatalogWriter<T> {
+
+    final static private String RESOURCE_INDEX_KEY = "resource.index";
+
+    final static private String CURRENT_RESOURCE_ITEM_COUNT = "resource.item.count";
+
+    private Resource resource;
+
+    private CatalogWriter<? super T> delegate;
+
+    private int itemCountLimitPerResource = Integer.MAX_VALUE;
+
+    private int currentResourceItemCount = 0;
+
+    private int resourceIndex = 1;
+
+    private ResourceSuffixCreator suffixCreator = new SimpleResourceSuffixCreator();
+
+    private boolean saveState = true;
+
+    private boolean opened = false;
+
+    public CatalogMultiResourceItemWriter(Class<T> clazz, Backup backupFacade,
+            XStreamPersisterFactory xStreamPersisterFactory) {
+        super(clazz, backupFacade, xStreamPersisterFactory);
+    }
+
+    protected void beforeStep(StepExecution stepExecution) {
+        delegate.setXp(getXp());
+    }
+
+    @Override
+    public void write(List<? extends T> items) throws Exception {
+        try {
+            if (!opened) {
+                File file = setResourceToDelegate();
+                // create only if write is called
+                file.createNewFile();
+                Assert.state(file.canWrite(),
+                        "Output resource " + file.getAbsolutePath() + " must be writable");
+                delegate.open(new ExecutionContext());
+                opened = true;
+            }
+            delegate.write(items);
+            currentResourceItemCount += items.size();
+            if (currentResourceItemCount >= itemCountLimitPerResource) {
```

Comment:
Wondering if the extra complication of this machinery is worth it, considering that in the app-context I see everywhere:

````
<property name="itemCountLimitPerResource" value="1" />
````